### ISSUE TITLE
Inconsistency between Rstudio Window and Browser Window

### ISSUE THREAD

**Issue by zachcp** [ramnathv/htmlwidgets#187]
Hi Team HTMLWidgets,

I've been making an HTMLWidget to [draw maps with d3](http://zachcp.github.io/quickd3maps-R/), (source code [here](https://github.com/zachcp/quickd3maps-R) ).  Maps render correctly with standard projections but trip an error  with non-standard projections (see linked webpage). This error typically means that dependencies were loaded in the incorrect order (d3.geo after d3) so I believe this is related to #179 . 

However, the confusing thing is that ALL maps render fine in RStudio - it is only when I go to the browser that I encounter an error.  Is the problem here that ordering is always an issue but Rstudio has a cache of all my JS?

I would appreciate any tips.

thanks,
zach cp

**Comment by ramnathv** [comment]
I have seen this error occur when there are non UTF8 characters in one of the javascript files, usually the case with unminified files. It shows up fine when the dependencies are not base64 encoded, which is why you are able to view it in RStudio, and it throws an error when you publish as selfcontained html.

I would try to make sure that all dependencies are UTF8 encoded, and then see if the error persists.

**Comment by zachcp** [comment]
Looks like it may be a UTF8 issue as the error trace  is:

```Uncaught SyntaxError: Unexpected token ILLEGAL```

**Comment by zachcp** [comment]
@ramnathv  thanks for the tip.

It looks like the error could be UTF8 related as the stacktrace is the following.
`Uncaught SyntaxError: Unexpected token ILLEGAL`

However, when I redownload the minified versions ( [as per this thread] (http://stackoverflow.com/questions/14080401/uncaught-syntaxerror-unexpected-token) ) to make sure there is no corruption I end up with the same set of errors. Is it possible the UTF-8 encoding is correct but  the encoding looks for something else, like ascii?

zach cp

**Comment by gordonwoodhull** [comment]
It's the projections javascript file in particular, search for `xB2` you'll see some weird stuff:

```js
                if (xB2 = b * b, dxBdÏ† = (2 * n + d * (4 * a + 6 * d * r)) * l, dyBdÏ† = e + d * (3 * o + 5 * d * i), dpdÏ† = 2 * (b * dxBdÏ† + p * (dyBdÏ† - 1)), dqdÏ† = 2 * (dyBdÏ† - 1), dmdÏ† = (dpdÏ† * q - w * dqdÏ†) / (q * q), cosÎ± = Math.cos(Q), sinÎ± = Math.sin(Q), mcosÎ± = m * cosÎ±, msinÎ± = m * sinÎ±, dÎ± dÏ† = v / pa * (1 / M(1 - xB2 / y)) * (dxBdÏ† * m - b * dmdÏ†) / y, fx = msinÎ± - u, fy = l * (1 + d * h) + m - mcosÎ± - s, Î´ xÎ´ Ï† = dmdÏ† * sinÎ± + mcosÎ± * dÎ± dÏ†, Î´ xÎ´ Î» = mcosÎ± * S, Î´ yÎ´ Ï† = 1 + dmdÏ† - (dmdÏ† * cosÎ± - msinÎ± * dÎ± dÏ†), Î´ yÎ´ Î» = msinÎ± * S, denominator = Î´ xÎ´ Ï† * Î´ yÎ´ Î» - Î´ yÎ´ Ï† * Î´ xÎ´ Î», !denominator) break;
```

found it by formatting it and then running
```
perl -ane '{ if(m/[[:^ascii:]]/) { print  } }'
```

**Comment by zachcp** [comment]
@ramnathv @gordonwoodhull. I just rebuilt the d3.geo.projections() file and it seems to work! thanks

**Comment by zachcp** [comment]
sorry. I wrote too soon. I'd changed to mercator. when I change back to naturalEarth I run into the same issue.

**Comment by zachcp** [comment]
Problem solved! It was a UTF encoding issue.

I'm not sure how the UTF corruption occurred, but occur it did.  I tried a few things I found on [this post](http://stackoverflow.com/questions/28396807/javascript-uncaught-syntaxerror-unexpected-token-illegal) and nothing seemed to work.  However, when I opened the file in TextMate and saved the file with the **add byte order mark** the resulting file could be encoded and viewed in the browser.

Thanks your feedback @ramnathv @gordonwoodhull 

<img width="541" alt="screen shot 2016-02-17 at 10 53 30 pm" src="https://cloud.githubusercontent.com/assets/1260667/13133454/5332bb60-d5c9-11e5-9bc1-93ef5a96540e.png">
